core/domain/fs_domain.py

```diff
@@ -471,6 +550,64 @@ def isfile(self, filepath):
         except cloudstorage.NotFoundError:
             return False
 
+    def get_file_content(self, filepath):  # pylint: disable=unused-argument
+        """Gets the content of the file stored in GCS
+        Args:
+            filepath: str. The path to the relevant file within the exploration.
+
+        Returns:
+            Content. The content of the file.
+        """
+        bucket_name = app_identity_services.get_gcs_resource_bucket_name()
+        gcs_file_url = (
+            '/%s/%s/assets/%s' % (
+                bucket_name, self._exploration_id, filepath))
+        gcs_file = cloudstorage.open(gcs_file_url)
+        contents = gcs_file.read()
+        gcs_file.close()
+
+        return contents
+
+    def get_image_dimensions(self, filepath):
```

Comment:
These have nothing to do with the file system, and should not be here.

You could either have an image_services file in core/domain, or call platform/ layer directly from the caller.

Reply:
Because we don't have get implemented for GCS. We directly fetch the file by the Public url for that entity `https://storage.googleapis.com/bucket_name ....`